### PR TITLE
Use httplib to get CCAMS working on non-Windows platforms (improved)

### PR DESCRIPTION
**Pull request by herver**
This PR builds on #48 but adds the following improvements:

 * Only use `httplib` when the `USE_HTTPLIB` symbol is set will default to Windows native client otherwise.
 * Added `Release-static` build target

I've also double checked that no unwanted dependency creeps in:

 * Dependencies of the `Windows native build`
```
>dumpbin /dependents .\Release\CCAMS.dll
Microsoft (R) COFF/PE Dumper Version 14.39.33523.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file .\Release\CCAMS.dll

File Type: DLL

  Image has the following dependencies:

    WININET.dll
    VERSION.dll
    EuroScopePlugInDll.dll
    KERNEL32.dll
    USER32.dll
    MSVCP140.dll
    VCRUNTIME140.dll
    api-ms-win-crt-runtime-l1-1-0.dll
    api-ms-win-crt-string-l1-1-0.dll
    api-ms-win-crt-heap-l1-1-0.dll
    api-ms-win-crt-convert-l1-1-0.dll
    api-ms-win-crt-stdio-l1-1-0.dll

  Summary

        2000 .data
        8000 .rdata
        2000 .reloc
        1000 .rsrc
       19000 .text
```

* Dependencies of the `httplib` (cross-platform) build:
```
>dumpbin /dependents .\Release-Static\CCAMS.dll
Microsoft (R) COFF/PE Dumper Version 14.39.33523.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file .\Release-Static\CCAMS.dll

File Type: DLL

  Image has the following dependencies:

    VERSION.dll
    EuroScopePlugInDll.dll
    KERNEL32.dll
    USER32.dll
    WS2_32.dll
    CRYPT32.dll
    ADVAPI32.dll

  Summary

        D000 .data
       9F000 .rdata
       1D000 .reloc
        1000 .rsrc
      267000 .text
```

### CODE REVIEW

**Comment by fagidutt** [comment]
while i can't say if the solution to use httplib is better overall, i wanted to mention that, regarding cross platform compatibility, while it is of course no true cross platform (which euroscope isn't, anyways), i successfully run Euroscope with all our plugins. With one non CCAMS-related exception, as AfV doesn't reliably work, so i use TrackAudio.
The solution for me to get CCAMS working in the prefix, was to install the vcrun2022 verb.

File: CCAMS.vcxproj [hunk 1]
```diff
@@ -35,11 +35,12 @@
     <PlatformToolset>v143</PlatformToolset>
     <PreferredToolArchitecture>x86</PreferredToolArchitecture>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release-Static|Win32'" Label="Configuration">
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release-static|Win32'" Label="Configuration">
     <ConfigurationType>DynamicLibrary</ConfigurationType>
     <CharacterSet>MultiByte</CharacterSet>
     <PlatformToolset>v143</PlatformToolset>
     <PreferredToolArchitecture>x86</PreferredToolArchitecture>
+    <UseOfMfc>false</UseOfMfc>
```
Comment (pierr3) [inline CCAMS.vcxproj:43]: This should be true if I am not wrong - ES SDK relies on MFC

**Comment by pierr3** [comment]
I would strongly push for default usage of httplib in any case, instead of the old Windows net libraries. While I also had success running this plugin with wine, I know this is not the case for every kind of distribution of linux. Using httplib greatly reduces the likelihood of such issues. If I had to re-write the ModeS/CCAMS plugin today, I would use httplib

File: CCAMS.vcxproj (hunk 1 above)
Comment (herver) [reply CCAMS.vcxproj]: Indeed, it should be fixed now, I cleaned up `AdditionalDependencies` to remove WinInet (not needed if we use `httplib`) but had to list a few others (namely `User32.lib`, `Advapi32.lib` and `Crypt32.lib`)

**Comment by kusterjs** [comment]
Is it possible that the httplib.h is missing? I'm unable to compile your version. Probably an issue on my side, as I'm very unfamiliar with the structure and processes of c++.

**Comment by herver** [comment]
> It might be worth just including httplib directly in the source tree since its header only would - remove any variables!
> […](#)

True, but `httplib` needs to be linked against OpenSSL to provide `HTTPS` support (see: https://github.com/yhirose/cpp-httplib?tab=readme-ov-file#ssl-support)

Which is why I decided it would be much easier to just use vcpkg instead of rebuilding openssl 😅

**Comment by pierr3** [comment]
Very good point, on windows linking OpenSSL is a pain so your way is
smarter.

**Comment by kusterjs** [comment]
@herver @pierr3 Thank you. Merged into separate branch for further improvements, if needed.

**Pull request merged**